addons/stock/models/stock_move.py

```diff
@@ -1023,7 +1005,13 @@ def _action_done(self):
             if move.state == 'cancel' or move.quantity_done <= 0:
                 continue
             moves_todo |= move
-            moves_todo |= move._create_extra_move()
+            moves_extra |= move._create_extra_move()
+
+        for move in moves_extra:
+            if move.exists():
+                move._action_confirm()
+
+        moves_todo = (moves_todo | moves_extra).exists()
```

Comment:
i don't like it :/ maybe we should just allow the merge between extra move and original move, and check after that if need more patches